app/build_images.go

```
@@ -0,0 +1,225 @@
+		StemcellImageID:      opt.StemcellID,
+		CompiledPackagesPath: f.StemcellCompilationDir(opt.Stemcell),
+		FissileVersion:       f.Version,
+	}
+
+	instanceGroups, err := f.Manifest.SelectInstanceGroups(opt.Roles)
+	if err != nil {
+		return err
+	}
+
+	if opt.OutputDirectory == "" {
+		err = f.buildPackagesImage(opt, instanceGroups, packagesImageBuilder)
+	} else {
+		err = f.buildPackagesTarball(opt, instanceGroups, packagesImageBuilder)
+	}
+	if err != nil {
+		return err
+	}
+
+	imageName, err := packagesImageBuilder.GetImageName(f.Manifest, instanceGroups, f)
+	if err != nil {
+		return err
+	}
+
+	roleImageBuilder := &builder.RoleImageBuilder{
+		BaseImageName:      imageName,
+		DarkOpinionsPath:   f.Options.DarkOpinions,
+		DockerOrganization: f.Options.DockerOrganization,
+		DockerRegistry:     f.Options.DockerRegistry,
+		FissileVersion:     f.Version,
+		Force:              opt.Force,
+		Grapher:            f,
+		LightOpinionsPath:  f.Options.LightOpinions,
+		ManifestPath:       f.Manifest.ManifestFilePath,
+		MetricsPath:        f.Options.Metrics,
+		NoBuild:            opt.NoBuild,
+		OutputDirectory:    opt.OutputDirectory,
+		RepositoryPrefix:   f.Options.RepositoryPrefix,
+		TagExtra:           opt.TagExtra,
+		UI:                 f.UI,
+		WorkerCount:        f.Options.Workers,
+	}
+
+	return roleImageBuilder.Build(instanceGroups)
+}
+
+// buildPackagesImage builds the docker image for the packages layer
+// where all packages are included
+func (f *Fissile) buildPackagesImage(
+	opt BuildImagesOptions,
+	instanceGroups model.InstanceGroups,
+	packagesImageBuilder *builder.PackagesImageBuilder,
+) error {
+
+	dockerManager, err := docker.NewImageManager()
+	if err != nil {
+		return fmt.Errorf("Error connecting to docker: %s", err.Error())
+	}
+
+	imageName, err := packagesImageBuilder.GetImageName(f.Manifest, instanceGroups, f)
+	if err != nil {
+		return fmt.Errorf("Error finding instance group's package name: %s", err.Error())
```

You don't need to propagate the error message with `err.Error()`, just `fmt.Errorf("whatever meaningful: %v", err)`.